lib/command_runner.dart

```diff
@@ -374,7 +374,7 @@ abstract class Command<T> {
   ///
   /// The return value is wrapped in a `Future` if necessary and returned by
   /// [CommandRunner.runCommand].
-  FutureOr<T> run() {
+  FutureOr<T>? run() {
```

Comment:
Why does this need to be nullable?

Reply:
This is also related to the design of how help is handled. I'm not super happy with the design but I don't see any easy wins for improving it. I'll likely just make a couple small tweaks.

Reply:
I'm having a hard time deciding between `FutureOr<T?>` and `FutureOr<T>?` here. The former would allow for a `Future<T?>` while the latter wont. Allowing that feels more consistent, but also unnecessary if we consider that the null return for the help command is the main use case. Do you have any opinion on that @jakemac53 ?

Reply:
I think that `FutureOr<T>?` is better.

The help command is a bit special because it isn't implemented by the user and we just can't really fix it without gross workarounds.

For user commands we should force them to return a valid value.